### PR TITLE
Banannon

### PR DESCRIPTION
**Pull request by Shadowmech88**
banannon_full
Banannon sprite courtesy of @Blithering.

Adds the banannon, a new weapon for traitor clowns. It can be bought for 18 TC.
The banannon fires armor-piercing discarding sabonanas, which deal 30 damage by default. In flight they drop their peel, on average about 4-5 tiles from the shooter, which increases their damage to 60 and spawns a banana peel on that tile.
The banannon holds 10 shots, and can be reloaded with bananas.
The clumsy check for the banannon is inverted, non-clumsy users have a 50% chance of the gun exploding in their hand.

:cl:
 * rscadd: Added the banannon, a new gun for traitor clowns.

### CODE REVIEW

**Comment by ghost** [comment]
>Giving the clown a banana rail gun
>That's purchasable using default traitor-level TCs

I know it's supposed to be an alternative to the revolver for clowns, but a 10 round magazine that actually does more damage the further away the target is, and can be reloaded using regular grown bananas

What the crap

**Comment by Shadowmech88** [comment]
What cost do you suggest?

**Comment by ghost** [comment]
Traitor max + the TC increase from being on the roundstart report

Only those clowns who are willing to go above and beyond may get the banannon

**Comment by Shadowmech88** [comment]
The average clown doesn't even have access to a steady supply of bananas, though.
It'd be tougher for them to refill it than for a traitor to refill a revolver.

**Comment by Aurx** [comment]
You can get bananas 9/10 rounds if you actually try.  
You can also get plant bags full of bananas.  Hundreds of bananas.  All in your backpack.  

As is, this is massively better than the revolver due to effortless access to ammo, larger magazine, and the creation of peels.

**Comment by ghost** [comment]
Ghetto botany?

Actual botany, then buy the banannon when you've got a full bag of nanas?

**Comment by Aurx** [comment]
Upon review of the code, this appears to be usable by non-clumsy individuals.  

Please rectify this most pressing issue, posthaste.

**Comment by Lokiamis** [comment]
60 damage after 4-5 tiles is fucked up

**Comment by Shadowmech88** [comment]
It's the same amount of damage that standard bullets deal at all distances.

**Comment by Kitfox88** [comment]
shadow the fact that you implemented my shitpost makes me happy

**Comment by SonixApache** [comment]
is it APFSDS or APFSDS-T?

**Comment by Ephx1** [comment]
There have literally been dozens of rounds where the botanists call me out of nowhere to give me a truckload of bananas. On top of that it's piss easy to grow your own with little to no effort.

**Comment by Kurfursten** [comment]
Doesn't the clown already have like four traitor items?

There are several jobs that have no specific traitor item - Chaplain, Paramedic, Trader, Virologist, Scientist, Captain, Head of Personnel...

**Comment by SonixApache** [comment]
IMO; keep it in, but for clownsquad bus shit rather than as a trator item.

**Comment by Shadowmech88** [comment]
* Banannon price increased to 18 TC
* The clumsy check for the banannon is now inverted, non-clumsy users have a 50% chance of the gun exploding in their hands.

**Comment by calzilla1** [comment]
Make it NODROP, play bannana man globally when bought,  and stop when the clown dies

File: code/modules/projectiles/guns/projectile/banannon.dm [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+		icon_state = "banannon_empty"
+	else
+		icon_state = "[initial(icon_state)]_[round((current_ammo/max_ammo) * 10)]"
+
+/obj/item/weapon/gun/banannon/examine(mob/user)
+	..()
+	to_chat(user, "<span class='info'>It has [current_ammo] round\s remaining.</span>")
+
+/obj/item/weapon/gun/banannon/afterattack(atom/A as mob|obj|turf|area, mob/living/user as mob|obj, flag, params, struggle = 0)
+	if(flag)
```
Comment (PJB3005) [inline code/modules/projectiles/guns/projectile/banannon.dm:40]: Or, you know, you're trying to point-blank somebody.

File: code/modules/projectiles/guns/projectile/banannon.dm [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+	else if(current_ammo <= 0)
+		icon_state = "banannon_empty"
+	else
+		icon_state = "[initial(icon_state)]_[round((current_ammo/max_ammo) * 10)]"
+
+/obj/item/weapon/gun/banannon/examine(mob/user)
+	..()
+	to_chat(user, "<span class='info'>It has [current_ammo] round\s remaining.</span>")
+
+/obj/item/weapon/gun/banannon/afterattack(atom/A as mob|obj|turf|area, mob/living/user as mob|obj, flag, params, struggle = 0)
```
Comment (PJB3005) [inline code/modules/projectiles/guns/projectile/banannon.dm]: No `as xxx`

File: code/modules/projectiles/guns/projectile/banannon.dm [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+		to_chat(user, "<span class='warning'>A label sticks the trigger to the trigger guard!</span>")//Such a new feature, the player might not know what's wrong if it doesn't tell them.
+
+		return
+	if(!current_ammo)
+		return click_empty(user)
+	if(in_chamber)
+		qdel(in_chamber)
+		in_chamber = null
+	in_chamber = new /obj/item/projectile/bullet/sabonana(src)
+	if(Fire(A,user,params, "struggle" = struggle))
```
Comment (PJB3005) [inline code/modules/projectiles/guns/projectile/banannon.dm]: BYOND even fucking *allows* named arguments to be passed as strings? Jesus

Remove the quotes.

File: code/modules/projectiles/guns/projectile/banannon.dm [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+		current_ammo--
+	update_icon()
+
+/obj/item/weapon/gun/banannon/Fire(atom/target as mob|obj|turf|area, mob/living/user as mob|obj, params, reflex = 0, struggle = 0, var/use_shooter_turf = FALSE)
```
Comment (PJB3005) [inline code/modules/projectiles/guns/projectile/banannon.dm]: Remove `as xxx`

File: code/modules/projectiles/guns/projectile/banannon.dm [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+	if(istype(user, /mob/living))
+		var/mob/living/M = user
+		if(!clumsy_check(M) && prob(50))
+			to_chat(M, "<span class='danger'>[src] blows up in your face.</span>")
```
Comment (PJB3005) [inline code/modules/projectiles/guns/projectile/banannon.dm]: `\the [src]`

File: code/modules/projectiles/guns/projectile/banannon.dm [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+			M.take_organ_damage(0,20)
+			M.drop_item(src, force_drop = 1)
```
Comment (PJB3005) [inline code/modules/projectiles/guns/projectile/banannon.dm]: Done by `qdel`, don't bother.

File: code/modules/projectiles/guns/projectile/banannon.dm [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+			qdel(src)
+			return 0
+	return ..()
+
+/obj/item/weapon/gun/banannon/process_chambered()
+	if(in_chamber)
```
Comment (PJB3005) [inline code/modules/projectiles/guns/projectile/banannon.dm]: so, `return in_chamber`?

File: code/modules/projectiles/guns/projectile/banannon.dm [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+	return 0
+
+/obj/item/weapon/gun/banannon/attackby(obj/item/W, mob/user)
+	..()
+	if(istype(W, /obj/item/weapon/reagent_containers/food/snacks/grown/banana))
+		if(current_ammo >= max_ammo)
+			return
+		current_ammo++
+		playsound(get_turf(src), 'sound/items/Deconstruct.ogg', 25, 1)
+		user.drop_item(W, force_drop = 1)
```
Comment (PJB3005) [inline code/modules/projectiles/guns/projectile/banannon.dm]: How would you insert a banana into the gun if the banana is stuck to your hand? Why force this?

**Review by PJB3005 — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/guns/projectile/banannon.dm (hunk 5 above)
Comment (Exxion) [reply code/modules/projectiles/guns/projectile/banannon.dm]: `\The`*

**Comment by Shadowmech88** [comment]
Conflicts fixed.

**Comment by yclatious** [comment]
https://www.youtube.com/watch?v=yModCU1OVHY

Banana man when

**Review by PJB3005 — APPROVED** [review]

**Pull request merged**